connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -150,23 +152,73 @@ public void setup() throws IOException {
         mm2Props.put("backup.bootstrap.servers", backup.kafka().bootstrapServers());
         mm2Config = new MirrorMakerConfig(mm2Props);
 
+        // we wait for the connector and tasks to come up for each connector, so that when we do the
+        // actual testing, we are certain that the tasks are up and running; this will prevent
+        // flaky tests where the connector and tasks didn't start up in time for the tests to be
+        // run
         backup.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorSourceConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorSourceConnector");
+        waitForTasksToBeUp(backup, "MirrorSourceConnector");
 
         backup.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorCheckpointConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorCheckpointConnector");
+        waitForTasksToBeUp(backup, "MirrorCheckpointConnector");
 
         backup.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorHeartbeatConnector.class));
+        waitForConnectorToBeUp(backup, "MirrorHeartbeatConnector");
+        waitForTasksToBeUp(backup, "MirrorHeartbeatConnector");
 
         primary.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorSourceConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorSourceConnector");
+        waitForTasksToBeUp(primary, "MirrorSourceConnector");
 
         primary.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorCheckpointConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorCheckpointConnector");
+        waitForTasksToBeUp(primary, "MirrorCheckpointConnector");
 
         primary.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorHeartbeatConnector.class));
+        waitForConnectorToBeUp(primary, "MirrorHeartbeatConnector");
+        waitForTasksToBeUp(primary, "MirrorHeartbeatConnector");
+
+    }
+
+    private void waitForConnectorToBeUp(EmbeddedConnectCluster connectCluster, String connName)
```

Comment:
If we let this take a set of connector names, we could test them all at once.

Reply:
Yes, this is correct, we could change this to take in a set of connector names, but as I mentioned in my previous comment, I separated it out for consistency sake with checking tasks. However, happy to change it to a set if that makes more sense.

Reply:
I'm mostly concerned about the time to run tests. If it does not make a substantial difference either way, then we can leave it.

Reply:
Yes, you have to check each task individually, but nothing prevents you from submitting them all at the same time. I am assuming that any overhead here is really associated with task startup time, so this allows for it to be parallelized.

Reply:
Fair enough, I'll change it to make the submission per connect cluster.